src/LiveComponent/assets/src/morphdom.ts

```diff
@@ -19,103 +19,126 @@ export function executeMorphdom(
-                    // apply the changes to the "to" element so it looks like the
-                    // external changes were already part of it
-                    elementChanges.applyToElement(toEl);
+                // Update child if parent has his live-id changed
+                if (fromEl.hasAttribute('parent-live-id-changed')) {
+                    fromEl.removeAttribute('parent-live-id-changed');
+
+                    return true;
                 }
 
-                // https://github.com/patrick-steele-idem/morphdom#can-i-make-morphdom-blaze-through-the-dom-tree-even-faster-yes
-                if (fromEl.nodeName.toUpperCase() !== 'OPTION' && fromEl.isEqualNode(toEl)) {
-                    // the nodes are equal, but the "value" on some might differ
-                    // lets try to quickly compare a bit more deeply
-                    const normalizedFromEl = cloneHTMLElement(fromEl);
-                    normalizeAttributesForComparison(normalizedFromEl);
+                // look for data-live-ignore, and don't update
+                return !fromEl.hasAttribute('data-live-ignore');
+            },
 
-                    const normalizedToEl = cloneHTMLElement(toEl);
-                    normalizeAttributesForComparison(normalizedToEl);
+            beforeNodeRemoved(node) {
+                if (!(node instanceof HTMLElement)) {
+                    // text element
+                    return true;
+                }
 
-                    if (normalizedFromEl.isEqualNode(normalizedToEl)) {
-                        // don't bother updating
-                        return false;
-                    }
+                if (externalMutationTracker.wasElementAdded(node)) {
+                    // this element was added by an external mutation, so we don't want to discard it
+                    return false;
                 }
-            }
 
-            // look for data-live-ignore, and don't update
-            return !fromEl.hasAttribute('data-live-ignore');
+                return !node.hasAttribute('data-live-ignore');
+            },
         },
+    });
 
-        onBeforeNodeDiscarded(node) {
-            if (!(node instanceof HTMLElement)) {
-                // text element
-                return true;
-            }
-
-            if (externalMutationTracker.wasElementAdded(node)) {
-                // this element was added by an external mutation, so we don't want to discard it
-                return false;
-            }
+    childComponentMap.forEach((childComponent, element) => {
+        const childComponentInResult = findChildComponent(childComponent.id ?? '', rootFromElement);
+        if (null === childComponentInResult || element === childComponentInResult) {
+            return;
+        }
 
-            return !node.hasAttribute('data-live-ignore');
-        },
+        childComponentInResult?.replaceWith(element);
+        childComponent.updateFromNewElementFromParentRender(childComponentInResult);
```

Comment:
Can you explain why this is now needed? I see some slight change in the logic above, but I can't quite understand the significance. Also, for reference, I'm checking the diff with https://github.com/symfony/ux/pull/1255/files?w=1 so that it hides whitespace differences and it's easier for me to see what *actually* changed.

Reply:
Ho man! This PR looks so far away now! But I think this is related to the last failing test. The thing is idiomorph is more clever than morphdom so when you have a component that has children and the children. If your component rerender and the children swap position idiopmorph is clever enough to just morph the position of the children. But the thing is this is so clever that the child is not rerender. So here we forced the children to be rerender in this situation. I hope this help